ticket_first_and_last_touch.view.lkml

```diff
@@ -0,0 +1,206 @@
+                      WHEN audits.metadata__system__location LIKE '%Canada%' THEN 'Canada'
+                      WHEN audits.metadata__system__location LIKE '%United Kingdom%' THEN 'UK'
+                      WHEN audits.metadata__system__location LIKE '%United Arab Emirates%' THEN 'UAE'
+                      ELSE 'Other' END) OVER (PARTITION BY ticket_id ORDER BY audits.created_at ASC) AS first_touch_agent_location,
+        FIRST_VALUE(users.name) OVER (PARTITION BY ticket_id ORDER BY audits.created_at ASC) AS first_touch_agent_name,
+        FIRST_VALUE(audits.created_at) OVER (PARTITION BY ticket_id ORDER BY audits.created_at ASC) AS first_touch_created_at,
+        FIRST_VALUE(CASE
+                      WHEN audits.metadata__system__location LIKE '%Philippines%' THEN 'Philippines'
+                      WHEN audits.metadata__system__location LIKE '%San Francisco%' THEN 'San Francisco'
+                      WHEN audits.metadata__system__location LIKE '%Mountain View%' THEN 'San Francisco'
+                      WHEN audits.metadata__system__location LIKE '%Englewood, CO%' THEN 'Mexico'
+                      WHEN audits.metadata__system__location LIKE '%Mexico%' THEN 'Mexico'
+                      WHEN audits.metadata__system__location LIKE '%United States%' THEN 'United States'
+                      WHEN audits.metadata__system__location LIKE '%Canada%' THEN 'Canada'
+                      WHEN audits.metadata__system__location LIKE '%United Kingdom%' THEN 'UK'
+                      WHEN audits.metadata__system__location LIKE '%United Arab Emirates%' THEN 'UAE'
+                      ELSE 'Other' END) OVER (PARTITION BY ticket_id ORDER BY audits.created_at DESC) AS last_touch_agent_location,
+        FIRST_VALUE(users.name) OVER (PARTITION BY ticket_id ORDER BY audits.created_at DESC) AS last_touch_agent_name,
+        FIRST_VALUE(audits.author_id) OVER (PARTITION BY ticket_id ORDER BY audits.created_at DESC) AS last_agent_id,
+        FIRST_VALUE(audits.created_at) OVER (PARTITION BY ticket_id ORDER BY audits.created_at DESC) AS last_touch_created_at
+      FROM
+        zendesk_stitch.ticket_audits AS audits
+      LEFT JOIN
+        zendesk_stitch.users AS users
+      ON
+        users.id = audits.author_id
+      LEFT JOIN
+        zendesk_stitch.ticket_audits__events as events
+      ON
+        events._sdc_source_key_id = audits.id
+      WHERE
+          (users.role = 'agent' OR users.role = 'admin')
+        AND
+          users.email != 'zendesk@getaround.com'
+        AND
+          ((events.public = true
+           AND
+          events.type = 'Comment')
+        OR
+          audits.via__channel = 'voice')
+       ;;
+    indexes: ["ticket_id"]
+    sql_trigger_value: SELECT COUNT(*) FROM zendesk_stitch.ticket_audits ;;
+  }
+
+ dimension: ticket_id {
+    primary_key: yes
+    type: number
+    hidden: yes
+    sql: ${TABLE}.ticket_id ;;
+  }
+
+  dimension: first_touch_agent_location {
```

Comment:
Thinking more about this:

- First Touch Agent Location
- First Touch Agent Group
- First Touch Agent ID

These feel like they offer maximum flexibility. We snapshot "Location" and "Group" since those can change, and then we pull in ID for all properties of the agent, including name? Tell me why not :)

Reply:
Agreed that these three offer the most flexibility.  Change has been made✅